src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
                 },
                 _joinableFactory);
 
+            _solutionLoadedEvent = new AsyncManualResetEvent(initialState: true);
```

Comment:
This is a change from your earlier iteration. Why are you *setting* this event here unconditionally? Before, your TaskCompletionSource was not completed, which would align with a `false` argument passed to `AsyncManualResetEvent`'s constructor. As it is, you're indicating that the solution is loaded at this point. Are you sure?